docs/src/tutorials/entry-tutorial.md

```diff
@@ -0,0 +1,1085 @@
+the feedback fast.
+
+Let us check `Postcondition` again:
+
+```sh
+$ apalache-mc check --cinit=ConstInit --inv=Postcondition MC5_8.tla
+...
+State 2: state invariant 0 violated. Check the counterexample in: 
+  /[a long path]/counterexample1.tla
+...
+```
+
+Let us inspect the counterexample:
+
+```tla
+---------------------------- MODULE counterexample ----------------------------  
+EXTENDS MC5_8
+
+(* Constant initialization state *)
+ConstInit == INPUT_KEY = -1 /\ INPUT_SEQ = <<0, -1>>
+
+(* Initial state *)
+State0 ==
+  INPUT_KEY = -1
+    /\ INPUT_SEQ = <<0, -1>>
+    /\ high = 1
+    /\ isTerminated = FALSE
+    /\ low = 0
+    /\ returnValue = 0
+
+(* Transition 1 to State1 *)
+State1 ==
+  INPUT_KEY = -1
+    /\ INPUT_SEQ = <<0, -1>>
+    /\ high = -1
+    /\ isTerminated = FALSE
+    /\ low = 0
+    /\ returnValue = 0
+
+(* Transition 3 to State2 *)
+State2 ==
+  INPUT_KEY = -1
+    /\ INPUT_SEQ = <<0, -1>>
+    /\ high = -1
+    /\ isTerminated = TRUE
+    /\ low = 0
+    /\ returnValue = -1
+...
+```
+
+**Fixing Postcondition.** Is it a real issue? It is, but it is not the issue of
+the search, rather our invariant `Postcondition` is imprecise.
+
+## Step 5b: Fixing the postcondition
+
+If we check our source of truth, that is, the Java documentation in
+[Arrays.java in OpenJDK][], we will see the following sentences:
+
+    The range must be sorted (as by the {@link #sort(int[], int, int)} method)
+    prior to making this call. If it is not sorted, the results are undefined.
+    If the range contains multiple elements with the specified value, there is
+    no guarantee which one will be found.
+
+It is quite easy to add this constraint. This is where TLA+ starts to shine:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch5.tla:79:88}}
+...
+{{#include ../../../test/tla/bin-search/BinSearch5.tla:109:113}}
+```
+
+If we check `PostconditionSorted`, we do not get any error after 10 steps:
+
+```sh
+$ apalache-mc check --cinit=ConstInit --inv=PostconditionSorted MC5_8.tla
+...
+The outcome is: NoError
+Checker reports no error up to computation length 10
+```
+
+It takes some time to explore all executions of length up to 10 steps, for all
+input sequences of length up to `2^7 - 1` arbitrary integers. If we think about
+it, the model checker managed to crunch infinitely many numbers in several
+hours. Not bad.
+
+*Exercise.* If you are impatient, you can check `PostconditionSorted` for the
+configuration that has integer width of 4 bits. It takes only a few seconds to
+explore all executions.
+
+TODO: add the link to BinSearch1.tla when we have it on github.
+
+## Step 6: Checking termination and progress
+
+Copy `BinSearch5.tla` to `BinSearch6.tla`, copy `MC5_8.tla` to `MC6_8.tla` and
+open `BinSearch6.tla`.
+
+Actually, we do not need 10 steps to check termination for the case `INT_WIDTH
+= 8`. If you recall the complexity of the binary search, it needs
+`log2(Len(INPUT_SEQ))` steps to terminate.
```

Comment:
You lost me here because I forgot that we let `INPUT_SEQ == Gen(MAX_INT)` and `MAX_INT == 2^(INT_WIDTH - 1) - 1`. Include a reminder?